From 87a8fd4dab0a94590ae80b14d2383d569a9fe45f Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 15 Oct 2014 21:42:00 -0400
Subject: [drm] vmwgfx: respect 'nomodeset'

Message-id: <1413409320-19272-1-git-send-email-rclark@redhat.com>
Patchwork-id: 97579
O-Subject: [PATCH] drm/vmwgfx: respect 'nomodeset'
Bugzilla: 1101381
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Rob Clark <robdclark@gmail.com>

RHBZ: 1101381
Upstream: sent, ack'd

Signed-off-by: Rob Clark <robdclark@gmail.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/vmwgfx/vmwgfx_drv.c b/drivers/gpu/drm/vmwgfx/vmwgfx_drv.c
index 246a62b..36a9d01 100644
--- a/drivers/gpu/drm/vmwgfx/vmwgfx_drv.c
+++ b/drivers/gpu/drm/vmwgfx/vmwgfx_drv.c
@@ -25,6 +25,7 @@
  *
  **************************************************************************/
 #include <linux/module.h>
+#include <linux/console.h>
 
 #include <drm/drmP.h>
 #include "vmwgfx_drv.h"
@@ -1460,6 +1461,12 @@ static int vmw_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 static int __init vmwgfx_init(void)
 {
  int ret;
+
+#ifdef CONFIG_VGA_CONSOLE
+ if (vgacon_text_force())
+  return -EINVAL;
+#endif
+
  ret = drm_pci_init(&driver, &vmw_pci_driver);
  if (ret)
   DRM_ERROR("Failed initializing DRM.\n");
-- 
1.7.1